### ISSUE TITLE
docs: add FAQ & Troubleshooting page

### ISSUE THREAD

**Issue by shinagawa-web** [shinagawa-web/gomarklint#153]
## Overview

Users run into common errors that are hard to debug without guidance. A dedicated FAQ/Troubleshooting page acts as a **hub** that routes users to the right deeper documentation — because FAQ is one of the pages people actually land on, while other docs pages are rarely visited proactively.

## Design principle

- Each entry must be a problem users **actually hit**, not a contrived question
- The answer gives a quick resolution inline
- Where relevant, the answer naturally leads to deeper docs — not as an explicit "read this doc" redirect, but woven into the explanation (e.g. "the full rule option reference is in [Configuration](../configuration/)")
- No obvious or patronizing questions

The FAQ's job: answer fast, and quietly hand off to the doc that goes deeper — without the user noticing it's happening.

## Problems to cover

Real examples of confusion:
- User has a v1 config (`enableLinkCheck`, `minHeadingLevel`, etc.) and doesn't understand why settings are ignored
- User sees `failed to parse config file` with no hint of what's wrong
- User on v1 sees `unknown field: default` because v2 introduced the `default` key
- No single place to look up what an error message means

## Proposed entries and routing targets

| Error / Situation | Quick fix | Routes to |
|---|---|---|
| Config options ignored / "unknown field" | v1 config used with v2 binary → `gomarklint init` | [Migrating to v2](../migration-v2/) |
| `failed to parse config file` | Invalid JSON or wrong rule value format | [Configuration](../configuration/) |
| `invalid rule value` | Use `true`, `false`, `"error"`, `"warning"`, or `"off"` | [Configuration](../configuration/) |
| `failed to access config file` | Wrong `--config` path or file missing | [CLI Reference](../cli/) |
| `unknown shorthand flag: 'v'` | Not yet supported, planned in #152 | — |
| `unknown flag: --enable-*` | v1 CLI flags removed in v2 | [Migrating to v2](../migration-v2/) |
| No violations reported | Wrong path / rule disabled / severity filter | [Configuration](../configuration/), [CLI Reference](../cli/) |
| Lint passes locally, fails in CI | Version mismatch or config not committed | [GitHub Actions Integration](../github-actions/) |
| External link check slow / timing out | Increase `timeoutSeconds` or add `skipPatterns` | [Configuration](../configuration/) |

Also add a link to the FAQ page in `README.md` and `README.ja.md` under the Documentation section.

**Comment by shinagawa-web** [comment]
## Implementation Plan

### Files to create
- `docs/content/docs/faq.md` — FAQ & Troubleshooting page

### Files to modify
- `README.md` — add FAQ link under Documentation section
- `README.ja.md` — add FAQ link under Documentation section

### Checklist
- [ ] Create `docs/content/docs/faq.md`
- [ ] Add link to README.md
- [ ] Add link to README.ja.md
- [ ] `make test-all`

**Comment by shinagawa-web** [comment]
## Addition: External link false-positive FAQ entries

Following the investigation in #147 (tracked in #155), two FAQ entries should be added to the page.

### New rows for the "Proposed entries" table

| Error / Situation | Quick fix | Routes to |
|---|---|---|
| External link check reports valid links as broken (403) | Go's default `User-Agent` is blocked by CDN-level bot protection (Cloudflare, Akamai). Add `"allowedStatuses": [403]` to opt out of treating 403 as an error, or add the domain to `skipPatterns`. | [Configuration](../configuration/), external-link limitations doc (#155) |
| Academic / journal links always flagged (doi.org, Wikipedia, etc.) | These sites aggressively block automated clients. `skipPatterns` is the most reliable workaround until #155 lands. Note that external link checking has inherent limits — the tool flags candidates, but final judgment stays with the author. | external-link limitations doc (#155) |

### Checklist additions (for the Implementation Plan comment)

- [ ] Add FAQ entry: external link 403 false-positives — cause, `allowedStatuses` workaround, `skipPatterns` workaround
- [ ] Add FAQ entry: academic/journal links always flagged — name specific site categories, explain bot-protection limitations, set expectation that some false-positives are unavoidable

---

## Addition: "Verify before filing" guidance

Add a note — either as a standalone FAQ entry or as a callout at the top of the external link section — asking users to confirm a link is genuinely unreachable before opening an issue.

**Suggested entry:**

| Error / Situation | Quick fix | Routes to |
|---|---|---|
| Not sure if a link is truly broken | Run the curl command below and check the status code gomarklint actually sees. If it returns 2xx or 3xx, the link is fine and the report is a false-positive — see the entries above. | — |

**Curl command to include in the FAQ:**

```sh
curl -s -o /dev/null -w "%{http_code}\n" \
  -A "Go-http-client/1.1" \
  https://your-url-here
```

- **2xx** — link is reachable; the report is a false-positive
- **3xx** — redirect; still reachable
- **403** — server is blocking automated clients (bot protection) — not a broken link
- **404** — link is genuinely broken
- **000 / connection error** — network issue or domain does not exist

Ask users to include the curl output when filing an issue about a suspected false-positive.

### Checklist addition

- [ ] Add "verify before filing" entry with curl command and status code reference table
